groovy/groovy.support/src/org/netbeans/modules/groovy/support/actions/singlefilerun/JPDAStart.java

```diff
@@ -0,0 +1,145 @@
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.util.RequestProcessor;
+import org.openide.windows.InputOutput;
+
+// This class is copy of org.netbeans.modules.java.api.common.singlesourcefile.JPDAStart
+/**
+ * Start the JPDA debugger.
+ *
+ * @author Arunava Sinha
+ */
+public class JPDAStart implements Runnable {
+
+    static final Logger LOG = Logger.getLogger(JPDAStart.class.getPackage().getName());
+    
+    private static final RequestProcessor RP = new RequestProcessor(JPDAStart.class);
+    private static final String TRANSPORT = "dt_socket"; //NOI18N
+
+    private final Object[] lock = new Object[2];
+    private final InputOutput io;
+    private final FileObject fileObject;
+
+    JPDAStart(InputOutput inputOutput, FileObject fileObject) {
+        io = inputOutput;
+        this.fileObject = fileObject;
+    }
+
+    /**
+     * returns the port that the debugger listens to..
+     */
+    public String execute() throws Exception {
+        LOG.log(Level.INFO, "JPDA Listening Start"); //NOI18N
+        synchronized (lock) {
+            RP.post(this);
+            lock.wait();
+            if (lock[1] != null) {
+                throw ((Exception) lock[1]); //NOI18N
+            }
+        }
+        return (String) lock[0];
+    }
+
+    @Override
+    public void run() {
+        synchronized (lock) {
+
+            try {
+
+                ListeningConnector lc = null;
+                Iterator i = Bootstrap.virtualMachineManager().
+                        listeningConnectors().iterator();
+                for (; i.hasNext();) {
+                    lc = (ListeningConnector) i.next();
+                    Transport t = lc.transport();
+                    if (t != null && t.name().equals(getTransport())) {
+                        break;
+                    }
+                }
+                if (lc == null) {
+                    throw new RuntimeException("No trasports named " + getTransport() + " found!"); //NOI18N
+                }
+
+                final Map args = lc.defaultArguments();
+                String address = lc.startListening(args);
+                try {
+                    int port = Integer.parseInt(address.substring(address.indexOf(':') + 1));
+                    Connector.IntegerArgument portArg = (Connector.IntegerArgument) args.get("port"); //NOI18N
+                    portArg.setValue(port);
+                    lock[0] = Integer.toString(port);
+                } catch (NumberFormatException e) {
+                    lock[0] = address;
+                }
+                LOG.log(Level.INFO, "Debug Port:{0}", lock[0]);  //NOI18N
+
+                final Map properties = new HashMap();
+
+                ClassPath sourcePath = ClassPathSupport.createClassPath(fileObject.getParent());
+                ClassPath jdkPath = ClassPathSupport.createClassPath(System.getProperty("java.class.path"));
```

Comment:
Strange. Why do we need classpath of the IDE? Things like `org-openide-util.jar` & co.?

Reply:
This is copy of the class for running Debugger in Java.

Reply:
.... hm, the `jdkPath` local seems to be unused anyway ?

Reply:
It is no there now.